src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java

```diff
@@ -50,22 +63,55 @@ public String getUrlForPath(String wikiPath) {
   }
 
   public String getUrlForWiki() {
-    return getUrlForWiki(Configuration.getWikiName());
+    return getUrlForWiki(Configuration.getWikiName(), false);
   }
 
   public String getUrlForWiki(String wikiName) {
-    String prefix = "wikia".equals(wikiName) ? "www." : "";
-    String suffix;
+    return getUrlForWiki(wikiName, false);
+  }
 
-    if (env.contains("dev") && !env.contains(SANDBOX_MERCURY_DEV)) {
-      String devBoxOwner = env.split("-")[1];
+  public String getUrlForWiki(boolean addWWW) {
+    return getUrlForWiki(Configuration.getWikiName(), addWWW);
+  }
 
-      suffix = "." + devBoxOwner + "." + "wikia-dev.com";
-    } else {
-      suffix = wikiName.endsWith("wikia") ? ".com" : ".wikia.com";
+  public String getUrlForWiki(String wikiName, boolean addWWW) {
+    EnvType env = Configuration.getEnvType(this.env);
+    final String wikiaName = isGlobal(wikiName);
+
+    String www = "";
+    if (addWWW) {
+      www = "www.";
     }
 
-    return composeUrl(prefix, wikiName, suffix);
+    switch (env) {
+      case DEV: {
+        String devBoxOwner = this.env.split("-")[1];
+        return String.format(DEV_URL_FORMAT, www, wikiaName, devBoxOwner, env.getWikiaDomain());
+      }
+      case PROD: {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested code block into a method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1199)